### ISSUE TITLE
Consume raising ConsumeException: Broker: Not leader for partition

### ISSUE THREAD

**Issue by Coder3333** [confluentinc/confluent-kafka-dotnet#935]
Description
===========

When I call the Consume method, if there are unread messages, it seems to work properly by returning a ConsumeResult, but if there are no messages, it throws "ConsumeException: Broker: Not leader for partition".  It will continue to throw this error every time I call Consume until a message gets published to the queue.  I have set EnablePartitionEof to false, so I expect Consume to just block indefinitely, not to throw exceptions.  How do I get it to stop throwing these errors?

Confluent.Kafka 1.0.0

How to reproduce
================

Set EnablePartitionEof to false and call Consume until there are no more messages left.

**Comment by mhowlett** [comment]
this is interesting, since I recently traced through the relevant parts of librdkafka as part of an effort to document error scenarios and didn't note that as a possible error (but there's a lot to it so i probably missed something).

more information would be helpful. is this is an existing topic with messages in it? if not is auto topic create enabled or disabled? how many partitions? how many brokers?

if you set the debug config property to all and provide logs here, that would be helpful.

**Comment by Coder3333** [comment]
I do have auto topic enabled and I am trying to consume a topic that already exists.  It just doesn't have any more messages in it.  Once the first message is published, consume does return the message.  This topic has 1 partition with 1 replica.  There are 3 brokers.

**Comment by mhowlett** [comment]
do you have a minimal complete application that can demonstrate this? (it would help a lot).

also, your logs have a whole bunch of librdkafka producer debug messages followed by consumer exceptions (are there consumer logs?).

i don't have a good idea off the top of my head what could be causing this, will need to investigate.

**Comment by Coder3333** [comment]
Yes, these logs are from a unit test, so there are both consumer and producer logs together.  Actually, I only see this error when I Consume from a unit tests, which is xUnit running .Net Core.  I do not have the problem from my ASP.Net Core application that uses the same client code and connects to the same server, which is 0.10.  I even had the Kafka cluster rebooted, but that didn't change anything.

**Comment by jmathai-confluent** [comment]
Please reopen if you are able to reproduce this outside of a unit test using a more recent version of the client.
